features/feature_case/dnn/cudnn-convp6.cu

```diff
@@ -0,0 +1,179 @@
+    cudaMemcpy(out, host_out.data(), sizeof(float) * on * oc * oh * ow, cudaMemcpyHostToDevice);
+    cudaMemcpy(filter, host_filter.data(), sizeof(float) * fk * fc * fh * fw, cudaMemcpyHostToDevice);
+
+    cudnnConvolutionDescriptor_t covdes;
+    cudnnCreateConvolutionDescriptor(&covdes);
+    cudnnSetConvolution2dDescriptor(covdes, 0, 0, 1, 1, 1, 1, CUDNN_CROSS_CORRELATION, CUDNN_DATA_FLOAT);
+
+    cudnnConvolutionFwdAlgoPerf_t perf_data;
+    int returned_count;
+    cudnnFindConvolutionForwardAlgorithm(handle, dataTensor, filterTensor, covdes, outTensor, 1, &returned_count, &perf_data);
+
+    size_t size;
+    void *workspacesize;
+    cudnnSetConvolutionMathType(covdes, CUDNN_FMA_MATH);
+    cudnnGetConvolutionForwardWorkspaceSize(
+        handle, 
+        dataTensor, 
+        filterTensor, 
+        covdes, 
+        outTensor, 
+        perf_data.algo, 
+        &size);
+    cudaMalloc(&workspacesize, size);
+
+    float alpha = 2.5f, beta = 1.5f;
+    cudnnConvolutionForward(
+        handle, 
+        &alpha, 
+        dataTensor, 
+        data, 
+        filterTensor, 
+        filter, 
+        covdes, 
+        perf_data.algo,
+        workspacesize, 
+        size, 
+        &beta,
+        outTensor, 
+        out);
+    cudaDeviceSynchronize();
+    cudaMemcpy(host_out.data(), out, sizeof(float) * on * oc * oh * ow, cudaMemcpyDeviceToHost);
+
+    std::vector<float> expect = {
+        17260, 17561.5, 17863, 18164.5,
+        18766, 19067.5, 19369, 19670.5,
+        20272, 20573.5, 20875, 21176.5,
+        21778, 22079.5, 22381, 22682.5,
+        
+        43204, 44145.5, 45087, 46028.5,
+        47910, 48851.5, 49793, 50734.5,
+        52616, 53557.5, 54499, 55440.5,
+        57322, 58263.5, 59205, 60146.5,
+        
+        69148, 70729.5, 72311, 73892.5,
+        77054, 78635.5, 80217, 81798.5,
+        84960, 86541.5, 88123, 89704.5,
+        92866, 94447.5, 96029, 97610.5,
+        
+        95092, 97313.5, 99535, 101756,
+        106198, 108420, 110641, 112862,
+        117304, 119526, 121747, 123968,
+        128410, 130632, 132853, 135074,
+
+        47356, 47657.5, 47959, 48260.5,
+        48862, 49163.5, 49465, 49766.5,
+        50368, 50669.5, 50971, 51272.5,
+        51874, 52175.5, 52477, 52778.5,
+        
+        137300, 138242, 139183, 140124,
+        142006, 142948, 143889, 144830,
+        146712, 147654, 148595, 149536,
+        151418, 152360, 153301, 154242,
+        
+        227244, 228826, 230407, 231988,
+        235150, 236732, 238313, 239894,
+        243056, 244638, 246219, 247800,
+        250962, 252544, 254125, 255706,
+        
+        317188, 319410, 321631, 323852,
+        328294, 330516, 332737, 334958,
+        339400, 341622, 343843, 346064,
+        350506, 352728, 354949, 357170,               
+        };
+    check(expect, host_out, expect.size(), 1.f);
+
+    cudnnDestroy(handle);
+    cudaFree(data);
+    cudaFree(out);
+}
+
+int main() {
+    test1<CUDNN_DATA_FLOAT>();
```

Comment:
Only the CUDNN_DATA_FLOAT was tested, will you plan to test other types?

Reply:
Currently, no plan to test other types.